tests/test_testRepo.py

```diff
@@ -126,15 +126,16 @@ def testAddDataIdValue(self):
         with self.assertRaises(ValueError):
             addDataIdValue(self.butler, "detector", 101, nonsenseField="string")
 
-        # Keywords imply different instruments
-        with self.assertRaises(RuntimeError):
-            addDataIdValue(self.butler, "exposure", 101, instrument="dummyCam", physical_filter="k2020")
-
-        # No skymap defined
-        with self.assertRaises(RuntimeError):
-            addDataIdValue(self.butler, "tract", 42)
-        with self.assertRaises(RuntimeError):
-            addDataIdValue(self.butler, "tract", 43, skymap="map")
+        # No skymap defined; should be created automatically.
+        addDataIdValue(self.butler, "tract", 42)
+        (tract_data_id,) = set(self.butler.registry.queryDataIds(["tract"]).limit(1))
+        self.assertEqual(tract_data_id["tract"], 42)
+        # And then a new one created here, since the now-existing one should
+        # have a random name that isn't 'map'.
+        addDataIdValue(self.butler, "tract", 43, skymap="map")
```

Comment:
Please do not allow this behavior; it will be too surprising for users. A user who specified a specific skymap is clearly assuming it exists.

Reply:
I think this is a case that would be addressed by permitting only implied dimension records to be created.